disnake/state.py

```diff
@@ -786,7 +786,7 @@ def parse_message_create(self, data: gateway.MessageCreateEvent) -> None:
         if channel:
             if channel.__class__ in (TextChannel, Thread, VoiceChannel):
                 channel.last_message_id = message.id  # type: ignore
-            if channel.__class__ is Thread:
+            if type(channel) is Thread and message.type is not MessageType.thread_starter_message:
```

Comment:
this should be `channel.__class__` to be consistent with the line above

Reply:
> this should be `channel.__class__` to be consistent with the line above

I was using `type()` coz pyright cry with `.__class__`